packages/react-reconciler/src/ReactFiberHydrationContext.js

```diff
@@ -460,7 +484,7 @@ function tryToClaimNextHydratableInstance(fiber: Fiber): void {
     // If we can't hydrate this instance let's try the next one.
     // We use this as a heuristic. It's based on intuition and not data so it
     // might be flawed or unnecessary.
-    nextInstance = getNextHydratableSibling(firstAttemptedInstance);
+    nextInstance = getNextHydratableAfterFailedAttempt();
```

Comment:
Something I don't get about how this used to work is

imagine in the dom there is a `<!-- $ -->` but we are trying to hydrate a `<div>`. The `tryHydrate` would fail and if you were not in a concurrent root the `getNextHydratableSibling(firstAttemptedInstance)` would return an element inside the suspense boundary.

Is there something else that would prevent this erroneous logic from occurring in the real world?